src/hardhat-deploy-ethers.ts

```diff
@@ -0,0 +1,59 @@
+/* Imports: External */
+import { Contract } from 'ethers'
+import { Provider } from '@ethersproject/abstract-provider'
+import { Signer } from '@ethersproject/abstract-signer'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+export const getDeployedContract = async (
+  hre: HardhatRuntimeEnvironment,
+  name: string,
+  options: {
+    iface?: string
+    signerOrProvider?: Signer | Provider | string
+  } = {}
+): Promise<Contract> => {
+  const deployed = await hre.deployments.get(name)
+
+  await hre.ethers.provider.waitForTransaction(deployed.receipt.transactionHash)
```

Comment:
Just make sure we're interacting with a contract that actually exists. Not sure if this is strictly necessary but seems better to be on the safe side.